docs/generic-parameters.md

```diff
@@ -0,0 +1,142 @@
+# Generic kind parameters
+
+Following parameters are common to many or all kinds of definition:
+
+* [External files - `<external_files`](#external-files---external_files)
+* [System packages - `<system_packages>`](#system-packages---system_packages)
+* [Healthcheck - `<healthcheck>`](#healthcheck---healthcheck)
+  * [`cmd` healthcheck](#cmd-healthcheck)
+  * [`fcgi` healthcheck](#fcgi-healthcheck)
+  * [`http` healthcheck](#http-healthcheck)
+
+#### External files - `<external_files>`
+
+This parameter allows you to download files through HTTP or HTTPS to put them
+at a specific path in your image. These files might be compressed or
+uncompressed, and you might either decompress a full archive or a single path
+from your archive. Moreover, it supports checksum verification.
+
+When merging this parameter with a parent stage, the list of external files 
+of the parent is appended to the child's list. As such, you can't remove a
+file from a parent declaration.
+
+```yaml
+external_files:
+  - url: <string>
+    compressed: <bool>
+    pattern: <string>
+    destination: <string>
+    checksum: <string>
+    mode: <octal_number>
+    owner: <string>
+```
+
+Example for an uncompressed file:
+
+```yaml
+external_files:
+  - url: https://github.com/NiR-/fcgi-client/releases/download/v0.2.0/fcgi-client.phar
+    destination: /usr/local/bin/fcgi-client
+    mode: 0750
+    owner: 1000:1000
+```
+
+Example for extracting a single file from an archive:
+
+```yaml
+external_files:
+  - url: https://blackfire.io/api/v1/releases/probe/php/linux/amd64/72
+    compressed: true
```

Comment:
there's a guessing of the compression type ?

Reply:
Indeed, I need to document them properly. For reference, here's the code responsible for unpacking the archive: https://sourcegraph.com/github.com/moby/buildkit@92f1fb55d5a96773ba094b53ecaa6a01a9877d39/-/blob/solver/llbsolver/file/unpack.go#L38:29

And supported formats are: tar + gzip, bzip2, xz.